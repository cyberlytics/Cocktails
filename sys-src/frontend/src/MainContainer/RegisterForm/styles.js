export const styles = {
    styleDiv : {
        position: "relative",
        height: "330px",
        width: "400px",
        top: "50px",
        left: "50%",
        transform: 'translate(-50%)',

        backgroundColor: 'rgba(0, 0, 0, 60%)',
        boxSizing: 'border-box',
    },
    styleInputField : {
        margin: '10px 10px',
        overflow: 'auto',
    },    
    styleButton : {
        backgroundColor: 'rgba(115, 194, 251, 80%)',
        marginTop: '5px',
        width: '99%',
        height: '50px',
        fontSize: '22px',
        padding: '0',
        border: 'none',
    },
    styleHeadline : {
        color: 'white',
        textAlign: 'center',
        paddingTop : '10px',
    },
    styleLinkText : {
        color: 'white',
        textAlign : 'center',
        fontSize: '21px',
    }
};