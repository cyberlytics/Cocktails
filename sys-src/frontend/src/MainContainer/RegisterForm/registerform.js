//Import modules
import React, { Component } from 'react';

//Import own UI-Elements
import IconTextField from '../../inputs/icontextfield';

//Import local ressources
import { styles } from './styles';
import { apiurl } from '../../api';

class RegisterForm extends Component {
    
    //Initialize username and password field with empty string and enable button
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            passwordValidation: '',
            buttonDisabled: false
        };
        this.onKeyUp = this.handleKeypress.bind(this);
    }

    //Make sure everything is accessible on the beginning
    componentDidMount() {
        this.resetForm();
    }

    //Reset form
    resetForm() {
        this.setState({
            username: '',
            password: '',
            passwordValidation: '',
            buttonDisabled: false
        })
    }

    //method to pass to childs, which enables them to set state for username, password or password validation
    setInputValue(property, val) {
        this.setState({
            [property] : val
        })
    }

    async handleSubmit() {
        if (!this.state.username) {
            return;
        }
        if (!this.state.passwordValidation) {
            return;
        }
        if (!this.state.password) {
            return;
        }

        //When username, password and password validation were provided, then disable button and proceed
        this.setState({
            buttonDisabled : true
        })

        try {
            let res = await fetch(apiurl + '/user/register', {
                method: "post",
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    username: this.state.username, 
                    password: this.state.password,
                    passwordValidation: this.state.passwordValidation,
                })
            });
            
            //process result to json
            let result = await res.json();

            //Request successful
            if (result && result.success) {
                alert("Registrierung erfolgreich");
                window.location.href='/login';
            }
            //if not successful then make alert with error text sent from api
            else if (result && result.success === false) {
                this.setState({buttonDisabled: false});
                alert(result.msg);
            }
        //error handling
        } catch (error) {
            this.resetForm();
            console.log(error.message);
        }
    }

    //Submit, when Enter Button was pressed
    handleKeypress = e => {
        if (e.key === "Enter") {
          this.handleSubmit();  
        }
    };

    render() { 
        return (
            <div style={styles.styleDiv} className="login-form" onKeyPress={this.handleKeypress}>
                <h1 style={styles.styleHeadline}>Register</h1>

                <div style={styles.styleInputField}>
                    <IconTextField type="username" name="usr" placeholder="Username" iconclass="fas fa-user" value={this.state.username ? this.state.username : ''} onChange={ (val) => this.setInputValue('username', val)} />
                </div>
                
                <div style={styles.styleInputField}>
                    <IconTextField type="password" placeholder="Password" className="password-input" iconclass="fas fa-lock" value={this.state.password ? this.state.password : ''} onChange={ (val) => this.setInputValue('password', val)} />
                </div>
                
                <div style={styles.styleInputField}>
                    <IconTextField type="password" placeholder="Password Validation" iconclass="fas fa-lock" value={this.state.passwordValidation ? this.state.passwordValidation : ''} onChange={ (val) => this.setInputValue('passwordValidation', val)} />
                </div>
                
                <div style={styles.styleInputField}>
                    <button style={styles.styleButton} variant="contained" type="submit" className="btn btn-primary" onClick={() => this.handleSubmit() } disabled={this.state.buttonDisabled}>Submit</button>
                </div>
            </div>
        );
    }
}
 
export default RegisterForm;