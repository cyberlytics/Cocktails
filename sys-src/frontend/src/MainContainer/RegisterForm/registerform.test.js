import React from 'react';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import TestUtils from 'react-dom/test-utils'

//Import own modules
import RegisterForm from './registerform';
import { expect, it } from '@jest/globals';


describe('Test for RegisterForm', () => {
    test("Test headline rendering", () => {
        const { getByText } = render(<RegisterForm/>);
        expect(getByText(/register/i)).toBeInTheDocument();
    });

    test("renders correctly Username", () => {
        const {queryByPlaceholderText} = render(<RegisterForm/>);
        const input = queryByPlaceholderText("Username");
        expect(input).toBeTruthy();
    });

    test("renders correctly Password", () => {
        const {queryByPlaceholderText} = render(<RegisterForm/>);
        const input = queryByPlaceholderText("Password");
        expect(input).toBeTruthy();
    });

    test("renders correctly Password Validation", () => {
        const {queryByPlaceholderText} = render(<RegisterForm/>);
        const input = queryByPlaceholderText("Password Validation");
        expect(input).toBeTruthy();
    });

    test("all IconTextFields rendered correctly", () => {
        const { getAllByTestId } = render(<RegisterForm/>);
        const nodes = getAllByTestId('icon-textfield');
        expect(nodes.length).toBe(3);
    });

    test("renders correctly Submit Button", () => {
        const { queryByText } = render(<RegisterForm/>);
        const button = queryByText("Submit");
        expect(button).toBeTruthy();
    });

    it('input of username field works', () => {
        const textComponent = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.scryRenderedDOMComponentsWithTag(textComponent, 'input')[0];
        TestUtils.Simulate.change(node, { target: { value: 'john'}});
        expect(textComponent.state.username).toEqual("john");
    });
    
    it('input of password field works', () => {
        const textComponent = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.scryRenderedDOMComponentsWithTag(textComponent, 'input')[1];
        TestUtils.Simulate.change(node, { target: { value: '12345'}});
        expect(textComponent.state.password).toEqual("12345");
    });

    
    it('input of password validation field works', () => {
        const textComponent = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.scryRenderedDOMComponentsWithTag(textComponent, 'input')[2];
        TestUtils.Simulate.change(node, { target: { value: '12345'}});
        expect(textComponent.state.passwordValidation).toEqual("12345");
    });

    it('submit button works', () => {
        const component = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "12345";
        component.state.passwordValidation = "12345";

        TestUtils.Simulate.click(node);

        //Form resetted
        expect(true).toBe(true);
    })

    it('submit with Enter key press works', () => {
        const component = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "12345";
        component.state.passwordValidation = "12345";

        TestUtils.Simulate.keyPress(node, { key : 'Enter'});

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    })

    it('no submit with empty username', () => {
        const component = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "";
        component.state.password = "12345";
        component.state.passwordValidation = "12345";

        TestUtils.Simulate.click(node);

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    });

    it('no submit with empty password', () => {
        const component = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "";
        component.state.passwordValidation = "12345";

        TestUtils.Simulate.click(node);

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    });

    it('no submit with empty password validation', () => {
        const component = TestUtils.renderIntoDocument(
            <RegisterForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "12345";
        component.state.passwordValidation = "";

        TestUtils.Simulate.click(node);

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    });
})