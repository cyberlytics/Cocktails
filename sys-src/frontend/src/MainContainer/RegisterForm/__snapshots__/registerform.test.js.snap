// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for RegisterForm no submit with empty password 1`] = `undefined`;

exports[`Test for RegisterForm no submit with empty password validation 1`] = `undefined`;

exports[`Test for RegisterForm no submit with empty username 1`] = `undefined`;

exports[`Test for RegisterForm submit with Enter key press works 1`] = `undefined`;
